addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/BridgeHandlerFactory.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.icloud.internal;
+
+import static org.openhab.binding.icloud.BindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.icloud.discovery.DeviceDiscovery;
+import org.openhab.binding.icloud.handler.BridgeHandler;
+import org.openhab.binding.icloud.handler.DeviceHandler;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link BridgeHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Patrik Gfeller - Initial contribution
+ */
+@Component(service = BridgeHandlerFactory.class, immediate = true, configurationPid = "binding.icloud")
+public class BridgeHandlerFactory extends BaseThingHandlerFactory {

```

Comment:
This name is a bit misleading - the class might better be called `ICloudHandlerFactory`.

Reply:
Done.